### ISSUE TITLE
Ephinea PSOBB crashes with add-on plugin and DXVK artifact build which addresses #3117

### ISSUE THREAD

**Issue by sodaboy581** [doitsujin/dxvk#4017]
When attempting to run this game with the add-on plugin with the latest fixed DXVK artifact build, the game immediately crashes.

The build fixes #3117 and was downloaded from https://github.com/doitsujin/dxvk/actions/runs/9179660999/artifacts/1524146257

The build works fine when not using the add-on plugin.

Normally, I'd say this is an add-on plugin problem, but it worked with the older DXVK version and it works with the other APIs like D3D8, D3D9, D3D11, and D3D12.  It just breaks with DXVK.

I'm including an apitrace so maybe we can figure out if this is a DXVK issue or an add-on plugin issue.

If you'd like to test, download the game from https://ephinea.pioneer2.net/ and install the plugin at https://github.com/Solybum/psobbaddonplugin/releases

### Software information
Ephinea: Phantasy Star Online Blue Burst

### System information
- GPU: NVIDIA RTX 4080
- Driver: 552.22
- Wine version:  
- DXVK version:  v2.3.1-53-g65dd3c7d
- https://github.com/doitsujin/dxvk/actions/runs/9179660999/artifacts/1524146257  (It's not an official release, it includes the fix from here: https://github.com/doitsujin/dxvk/actions/runs/9179660999 )

### Apitrace file(s)
- https://files.pioneer2.net/PsoBB-trace.zip

Thanks for any insight you can provide!

**Comment by Blisto91** [comment]
I tried using the addon on Linux through Proton with a AMD GPU and couldn't get it to crash. But will try also on Windows.

**Comment by sodaboy581** [comment]
> I tried using the addon on Linux through Proton with a AMD GPU and couldn't get it to crash. But will try also on Windows.

Yep, you're right.  We have had a couple of people test on Linux and it appears fine there with Proton or WINE.

This appears to be a Windows only issue.  It's weird that this happens when the addon is used with the new DXVK under Windows.  I know the addon loads through dinput8.dll... Does DXVK use parts of DirectInput?

I'd love to provide more data to you, but the Event Viewer log when the game crashes just has some bogus offset data in it.

Example:

Faulting application name: PsoBB.exe, version: 0.0.0.0, time stamp: 0x00000000
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0xffff06fb
Faulting process id: 0x0x4C30
Faulting application start time: 0x0x1DAAC80C859B4A3
Faulting application path: C:\Users\Terry\EphineaPSO\PsoBB.exe
Faulting module path: unknown
Report Id: efed9e40-fb2b-4e07-bc25-123be82fdd54
Faulting package full name: 
Faulting package-relative application ID: 

That fault offset is crazy.  So I couldn't look at the code in memory to attempt to help debug.

Another time it was 0xffff132d, another time 0xffff07d9, and another time 0xffff1327.

The offset it crashes on isn't useful at all.

I've tried compiling the game's DLL without VMPROTECT as well to see if VMP hated DXVK + addon combo, but that didn't help at all.  Still crashed.

I've even tried changing the addon initialization LUA to do nothing, but it doesn't even appear we're getting to the part where the addon tries to initialize.  Crash seems to be happening way before.

**Comment by Blisto91** [comment]
Might be worth trying with https://github.com/doitsujin/dxvk/pull/4026

**Comment by WinterSnowfall** [comment]
> I know the addon loads through dinput8.dll... Does DXVK use parts of DirectInput?

It does not. However, a lot of mods/game fixes will load using dinput8.dll and override various unrelated method calls. Or, better said, dinput(8) is a common way to inject custom code into a process. Without access to the source code of the plugin it's rather hard to tell if it's messing with anything dxvk would care about.

**Comment by sodaboy581** [comment]
Hello!  Sorry for the late reply.  I've had a busy day and finally got around to setting up an environment to compile DXVK and add the changes from https://github.com/doitsujin/dxvk/pull/4026 into it.  I also got the quick commit that was done today for "Fix strange type in dynamic vertex upload"... ANYHOW, compiling all that, with both files updated, the addon still crashes in Windows with DXVK.  :shrug:

It's not a big deal, I suppose people have other API choices like D3D8, 9 (through 8to9), 11 and 12 (through dgVoodoo) in Windows for PSOBB that work with the addon, so not a big deal.  Seems fine on Linux which is probably what DXVK is primarily for anyhow.

Maybe someday it'll be figured out... and then maybe not.  Probably will have to work with the addon devs on this one.

Anyhow, thanks for your input!

**Comment by Blisto91** [comment]
I've tried on Windows now but still haven't gotten it to crash

<details>
  <summary>Screenshot</summary>

![image](https://github.com/doitsujin/dxvk/assets/47954800/7f52a463-e382-48ba-83a9-b994fac9bbea)
</details>

Do you apply any specific settings?
Also were the other Windows testers also using Nvidia?

**Comment by Blisto91** [comment]
Switching to my Nvidia GPU it still doesn't crash for me on Linux, but i can reproduce that it crashes on Windows. But it does that together with every dxvk version I've tried when used together with the addon, not just master. AMD iGPU runs it fine.

**Comment by sodaboy581** [comment]
I am SO sorry for not commenting on the previous post.  I was busy when I read it and forgot to circle back.

So, this seems more like an NVIDIA bug with their Vulkan implementation on Windows.

**Comment by sodaboy581** [comment]
Woops, didn’t mean to close.  Haha

**Comment by Blisto91** [comment]
@sodaboy581 Has there been any change in regards to this issue?

**Comment by Blisto91** [comment]
@sodaboy581 Friendly ping
